### ISSUE TITLE
PySpark interface

### ISSUE THREAD

**Issue by zommerfelds** [saurfang/spark-knn#11]
Do you think it is would be easy to add a Python interface for this module?
I might try to do this for my research but I am not sure yet what is the best way to support Python. How would the code be sent to Spark? My best idea is that it would require to add a Python egg with a wrapper of this module and a jar with the Scala source to the spark-submit command (since Python users usually don't use SBT).

**Comment by saurfang** [comment]
Since I tried to rigorously follow spark.ml, I believe PySpark interface is not too hard to have. You can check out upstream Spark project for reference. For example: https://github.com/apache/spark/blob/branch-1.6/python/pyspark/ml/classification.py

**Comment by zommerfelds** [comment]
I started adding the interface for `ml.classification` at <https://github.com/saurfang/spark-knn/compare/master...zommerfelds:master>.
I had to make some changes to the predictor because the Python module was always sending the parameter `bufferSize = -1` as a user defined param and it would propagate to the model by overwriting the `tau` that was first passed to the model.
Feel free to make any comments on my changes. If you like it, I can add a similar interface to the `ml.regression` part as well and then I can submit a PR.
